include/pybind11/pybind11.h

```diff
@@ -1322,7 +1322,7 @@ PYBIND11_NAMESPACE_BEGIN(detail)
 
 template <>
 struct handle_type_name<module_> {
-    static constexpr auto name = const_name("module");
+    static constexpr auto name = const_name("types.ModuleType");
```

Comment:
Could you please add a test? I think it's really simple:

Add in tests/test_modules.py:

```
def test_module_handle_type_name():
    assert m.def_submodule.__doc__ == "TODO"
```
or possibly
```
def test_module_handle_type_name(doc):
    assert doc(m.def_submodule) == "TODO"
```
I'd try the `__doc__` version first because it's stricter.

Reply:
I have added this test